lambdas/service/app.py

```diff
@@ -341,7 +345,17 @@ def service_config(self) -> ServiceConfig:
 
     @property
     def facets(self) -> Sequence[str]:
-        return sorted(self.service_config.translation.keys())
+        return list(self.field_types_by_filterable_facets.keys())
+
+    @cached_property
+    def field_types_by_filterable_facets(self) -> Mapping[str, FieldType]:
+        field_types_by_filterable_facets = dict()
+        for facet, location in self.service_config.translation.items():
+            field_type = self.repository_controller.service.field_type(app.catalog,
+                                                                       tuple(location.split('.')))
+            if isinstance(field_type, FieldType):
+                field_types_by_filterable_facets[facet] = field_type
+        return dict(sorted(field_types_by_filterable_facets.items()))
```

Comment:
Why is this sorted? Where is the non-determinism?

Reply:
The `ServiceApp.facets` property is already sorted because it's used in documentation (e.g. `facets` in https://service.dev.singlecell.gi.ucsc.edu/#/Index/get_index_samples)

Rather than sorting the values at the `facet` property I opt to sort the dictionary so the callers don't need to. 

Alternatively we can alpha sort the `ServiceConfig.translations`

Reply:
During parking lot we decided to have the facets presented in documentation in in the order that they are defined within the translations, this retains some grouping by entity type.